src/tasks/Tasks.ts

```diff
@@ -0,0 +1,869 @@
+            //   [
+            //     ...this.tasksQueuedDbPath,
+            //     utils.lexiPackBuffer(taskData.priority),
+            //     ...kP
+            //   ],
+            //   taskIdBuffer,
+            //   true
+            // );
+            await this.queueTask(
+              taskId,
+              taskScheduleTime,
+              taskData.priority
+            );
+          }
+
+          // When the transaction commits
+          // trigger the queue
+          // tran.queueSuccess(() => {
+          //   this.triggerQueuing();
+          // });
+
+          // Get the next task to be scheduled and set the timer accordingly
+          let nextScheduleTime: number | undefined;
+          for await (const [kP] of tran.iterator(
+            this.tasksScheduledDbPath,
+            { limit: 1, values: false }
+          )) {
+            nextScheduleTime = utils.lexiUnpackBuffer(kP[0] as Buffer);
+          }
+          if(abortController.signal.aborted) return;
+          if (nextScheduleTime == null) {
+            this.logger.debug('Scheduling loop iteration found no more scheduled tasks');
+          } else {
+            this.triggerScheduling(nextScheduleTime);
+          }
+          this.schedulerLogger.debug('Finish scheduling loop iteration');
+        });
+      }
+    })();
+    this.schedulingLoop = PromiseCancellable.from(
+      schedulingLoop,
+      abortController
+    );
+    this.schedulerLogger.info('Started Scheduling Loop');
+  }
+
+  protected async startQueueing() {
+    if (this.queuingLoop != null) return;
+    this.queueLogger.info('Starting Queueing Loop');
+    const abortController = new AbortController();
+    const abortP = utils.signalPromise(abortController.signal);
+    const queuingLoop = (async () => {
+      while (!abortController.signal.aborted) {
+        try {
+          await Promise.race([this.queuingLock.waitForUnlock(), abortP]);
+        } catch (e) {
+          if (e === abortQueuingLoopReason) {
+            break;
+          } else {
+            throw e;
+          }
+        }
+        this.queueLogger.debug(`Begin queuing loop iteration`);
+        [this.queuingLockReleaser] = await this.queuingLock.lock()();
+        await this.db.withTransactionF(async (tran) => {
+          for await (const [kP, taskIdBuffer] of tran.iterator(
+            this.tasksQueuedDbPath,
+          )) {
+            if(abortController.signal.aborted) break;
+            // Remember the activePromises isn't being incremented
+            if (this.activePromises.size >= this.activeLimit) break;
+            // Remove task from the queued index
+            await tran.del([...this.tasksQueuedDbPath, ...kP]);
+            const taskId = IdInternal.fromBuffer<TaskId>(taskIdBuffer);
+            const taskData = (await tran.get<TaskData>([...this.tasksTaskDbPath, taskIdBuffer]))!;
+            this.queueLogger.debug(`Queueing loop iteration starting Task ${taskId.toMultibase('base32hex')}`);
```

Comment:
If you're converting to `base32hex` you can use the utility function for that.

```ts
this.queueLogger.debug(`Queueing loop iteration starting Task ${tasksUtils.encodeTaskId(taskId)}`);
```

`decodeTaskId` also exists but I don't think it's used anymore since you're using `TaskString` for the events and maps.

Reply:
Yes should be converted to `encodeTaskId :: TaskId -> TaskIdEncoded`. The `TaskIdString` can remain an implementation detail.

Reply:
Doing this now. And I realise since I'm converting it, I might as well keep using `TaskIdEncoded`, and not bother with `TaskIdString`. Might make it easier to debug to maintain parity here anyway.

Reply:
I believe this has been resolved?